examples/pytorch/pyg/main.py

```diff
@@ -0,0 +1,188 @@
+        edge_index = (
+            torch.Tensor(srcs).long(),
+            torch.Tensor(edge[0]).long(),
+        )
+        return edge_index
+
+    def get_all_edge_attrs(self):
+        """Obtain all edge attributes stored in the :class:`GraphStore`."""
+        output = []
+        ta = EdgeAttr(("0", "0", "0"), "coo", size=[self.ge.node_count(0), 2708])
+        output.append(ta)
+
+        return output
+
+
+def train():
+    """Train the model."""
+    model.train()
+
+    total_loss = total_examples = 0
+    for data in tqdm.tqdm(loader):
+        data = data.to(device)
+        data = data[("0", "0", "0")]
+        optimizer.zero_grad()
+
+        h = model(data.x, data.edge_index)
+
+        h_src = h[data.edge_label_index[0]]
+        h_dst = h[data.edge_label_index[1]]
+        link_pred = (h_src * h_dst).sum(dim=-1)  # Inner product.
+
+        loss = F.binary_cross_entropy_with_logits(link_pred, data.edge_label)
+        loss.backward()
+        optimizer.step()
+
+        total_loss += float(loss) * link_pred.numel()
+        total_examples += link_pred.numel()
+
+    return total_loss / total_examples
+
+
+def test():
+    for data in tqdm.tqdm(loader):
+        data = data[("0", "0", "0")]
+        pred = model(data.x, data.edge_index)
+        pred = pred.argmax(1).detach().numpy()
+        test_f1 = f1_score(data.y, pred, average="micro")
+        return test_f1
+
+
+if __name__ == "__main__":
+    ge = Cora()
+    loader = LinkNeighborLoader(
+        (
+            DeepGNNFeatureStore(ge),
+            DeepGNNGraphStore(ge, f"{ge.data_dir()}/train.nodes"),
+        ),
+        batch_size=140,
+        shuffle=True,
+        neg_sampling_ratio=1.0,
+        num_neighbors=[5, 5],
+        num_workers=1,
+        persistent_workers=True,
+        edge_label_index=(
+            ("0", "0", "0"),
+            torch.Tensor(ge.sample_edges(2708, np.array(0), SamplingStrategy.Random))
+            .long()[:, :2]
+            .T,
+        ),
+    )
+
+    test_loader = LinkNeighborLoader(
+        (DeepGNNFeatureStore(ge), DeepGNNGraphStore(ge, f"{ge.data_dir()}/test.nodes")),
+        batch_size=140,
+        shuffle=True,
+        neg_sampling_ratio=1.0,
+        num_neighbors=[5, 5],
+        num_workers=1,
+        persistent_workers=True,
+        edge_label_index=(
+            ("0", "0", "0"),
+            torch.Tensor(ge.sample_edges(2708, np.array(0), SamplingStrategy.Random))
+            .long()[:, :2]
+            .T,
+        ),
+    )
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    model = GraphSAGE(
+        in_channels=121,
+        hidden_channels=64,
+        num_layers=2,
+        out_channels=7,
+    ).to(device)
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.005)
+
+    for epoch in range(1, 10):
+        loss = train()
+        test_f1 = test()
+        print(f"Epoch: {epoch:03d}, Loss: {loss:.4f}, Test F1: {test_f1:.4f}")
+    assert loss <= 0.55
```

Comment:
You want to assert f1 value on test dataset, not loss.

Reply:
See other comment above, I need help with this.